### ISSUE TITLE
Google contacts photos not being saved

### ISSUE THREAD

**Issue by steven989** [timsly/carrierwave-data-uri#13]
Hi, I'm creating an app that's trying to download Google Contacts photos and save them to S3. Google contacts API streams the binary data directly, a code like this

               image_file = open('profilephoto.jpg', 'wb')
               image_file.write(photo_object[:body])
               image_file.close()

literally saves the image file to the local directory. (And this works, I can see the saved photo locally)

Where I'm having issue is when I attempted to use this gem to save the streamed contacts photo binary data to S3 like this:

```
new_connection.photo_data_uri = "data:#{photo_object[:content_type]};base64,#{encoded}"
new_connection.save
```

Two issues pop up:
1) If in my photo_uploader.rb file I have Rmagick processing (like resize), Rmagick throws an error with "Photo missing translation"
2) If I remove the processing, no error comes up but the file saved onto S3 is empty. 

I'm not sure what's going on. Any suggestions? 

Thanks.

**Comment by timfjord** [comment]
Hi @steven989 

I am using this gem with S3 so I can ensure it is working fine with it.
So probably there are some issues in your code.
First, you need to ensure that you pass valid data uri. The best way to check that is to pass value of this `"data:#{photo_object[:content_type]};base64,#{encoded}` to, say, img src attribute(`<img width="16" height="16" alt="star" src="<%= "data:#{photo_object[:content_type]};base64,#{encoded}" %>" />`) and if data is valid you will see image.
Then you can move to ActiveRecord side.
On AR side you can first try with local storage and ensure you can save data there. Then switch to S3.

To be honest, this `Photo missing translation` message seems came from some other place. Seems you have locale different from `:en` and it is trying to translate error message. You can check error directly by inspecting `new_connection.errors`

**Comment by steven989** [comment]
Thanks for the quick reply. After digging around I think I found the issue. 

The encoding method that I used to turn the binary data I got from Google into base64, `Base64.encode`, inserts \n linefeed character after every 60 characters. Because of this the parsing regex `/^data:(.*?);(.*?),(.*)$/` in the gem only took photo data up until the first \n and ignored the rest of the data, which is why all the photos saved onto S3 in my previous tries were blanks.

The solution turned out to be using the `Base64.strict_encode64` method to encode instead. All good now. Thanks.
